react/Alerter/index.jsx

```diff
@@ -1,19 +1,94 @@
 import React, { Component } from 'react'
 import classNames from 'classnames'
+import { connect } from 'react-redux'
+
+import { translate } from '../helpers/i18n'
 
 import styles from './index.styl'
 
+const ALERT_SHOW = 'ALERT_SHOW'
+const ALERT_DISMISS = 'ALERT_DISMISS'
+const ALERT_CLEAR = 'ALERT_CLEAR'
+
 const DEFAULT_AUTOCLOSE_DELAY = 3500
+const DEFAULT_ALERT_LEVEL = 'info'
+const DISMISS_DELAY = 500
+
+const reducer = (state = [], action) => {
+  if (action.alert) {
+    if (!action.alert.id) {
+      action.alert.id = new Date().getTime()
+    }
+    return [action.alert, ...state]
+  }
+  switch (action.type) {
+    case ALERT_SHOW:
+      return [action.alert, ...state.filter(a => a.id !== action.id)]
+    case ALERT_DISMISS:
+      return state.filter(a => a.id !== action.id)
+    case ALERT_CLEAR:
+      return []
+    default:
+      return state
+  }
+}
+
+export default reducer
+
+// Action creators
+export const alertShow = (message, messageData = null, level = DEFAULT_ALERT_LEVEL) => ({
+  type: ALERT_SHOW,
+  id: new Date().getTime(),
+  message,
+  messageData,
+  level
+})
 
-class Alerter extends Component {
+export const alertDismiss = id => ({
+  type: ALERT_DISMISS,
+  id
+})
 
+export const alertClear = () => ({
+  type: ALERT_CLEAR
+})
+
+// Dumb component that wraps the alerts
+const Wrapper = ({ t, alerts, dismiss }) => (
+  <div className={styles['coz-alerter']}>
+    {alerts.map(alert =>
```

Comment:
Yes, I think it'd be useful to be able to display multiple alerts at the same time, but you're right, the CSS can't handle it right now. That's a job for @GoOz :smile: